jerry-core/debugger/jerry-debugger.c

```diff
@@ -1 +1,101 @@
+#include "jerry-port.h"
+
+#define PORT 5001
+#define BACKLOG 1
+
+int sock;                     /**< socket file descriptor for the remote communication */
+int connected;                /**< hold the file descriptor for the accepted socket */
+
+struct sockaddr_in server_addr, client_addr;  /**< declarations of the socket address */
+/** it is contain the size of the structure pointed by server_addr and client_addr */
+socklen_t sin_size = sizeof (struct sockaddr_in);
+
+bool remote_init ()
+{
+  bool optval = true;    /**< the arguments optval and optlen are used to access option values for setsockopt() */
+
+  /* Server adress declaration */
+  server_addr.sin_family = AF_INET;             /**< the address family is host by order*/
+  server_addr.sin_port = htons (PORT);           /**< host to network long (PORT) */
+  server_addr.sin_addr.s_addr = INADDR_ANY;     /**< ip address */
+  bzero (&(server_addr.sin_zero), BACKLOG);     /**< sets the first byte of the area starting with zero */
+
+  /* Create an endpoint for communication */
+  if ((sock = socket (AF_INET, SOCK_STREAM, 0)) == -1)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Socket error!\n");
+    return false;
+  }
+
+  /* Set the options on socket */
+  if (setsockopt (sock, SOL_SOCKET, SO_REUSEADDR, &optval, sizeof (int)) == -1)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Setsockopt error!\n");
+    return false;
+  }
+
+  /* Bind to the server address */
+  if (bind (sock, (struct sockaddr *)&server_addr, sizeof (struct sockaddr)) == -1)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Bind error, unable to bind!\n");
+    return false;
+  }
+
+  /* Listen for connections on socket */
+  if (listen (sock, BACKLOG) == -1)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Listen error!\n");
+    return false;
+  }
+
+  jerry_port_log (JERRY_LOG_LEVEL_DEBUG, "Waiting for the client connection.\n");
+
+  /* Connect from the client */
+  connected = accept (sock, (struct sockaddr *)&client_addr, &sin_size);
+
+  jerry_port_log (JERRY_LOG_LEVEL_DEBUG,"Connected from: %s:%d\n",
+                  inet_ntoa (client_addr.sin_addr),ntohs (client_addr.sin_port));
+
+  return true;
+} /* remote_init */
+
+/* Close the socket connection with the client */
+void connection_closed ()
+{
+  jerry_port_log (JERRY_LOG_LEVEL_DEBUG, "TCPServer connection closed on port %d.\n", PORT);
+  close (connected);
+  close (sock);
+} /* connection_closed */
+
+
+/* Send the parsed file names to the client side */
+void send_to_client (const char *data, uint16_t data_len)
+{
+  ssize_t byte_send;  /**< hold the ammount of how many bytes we send */
+
+  byte_send = send (connected, data, data_len, 0);
+
+  /** Check the byte send size with the data because there may be if the
+  * package is too big, the socket can't able to send throw the connection.
+  */
+  if (byte_send != (ssize_t) data_len)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error, there is some missing package...\n");
```

Comment:
There is a `byte_send` which has the return value of send ( on success, these calls return the number of bytes sent ). I check the `data_len` with this, and if the two value is not equal there is some missing bytes. Maybe I need to add to check the -1 ( error ), what do you think?